### PR TITLE
Fix interactive story counter

### PR DESCRIPTION
**Pull request by rafaelteodosio**
**Proposed changes**:
- Fix bug Interactive story counter doesn't update
- Resolve [#7462](https://github.com/RasaHQ/rasa/issues/7462)

**Status**:
- [x] added some tests for the functionality
- [ ] updated the documentation
- [x] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/main/changelog) for instructions)
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

File: rasa/core/training/interactive.py [hunk 1]
```diff
@@ -75,6 +75,8 @@
 
 MAX_VISUAL_HISTORY = 3
 
+
```
Comment (arthurTemporim) [inline rasa/core/training/interactive.py:78]: This is made according to style sheet?

File: rasa/core/training/interactive.py [hunk 2]
```diff
@@ -816,10 +818,15 @@ def _write_stories_to_file(
     else:
         append_write = "w"  # make a new file if not
 
+    with open(
+        export_story_path, "r", encoding=rasa.shared.utils.io.DEFAULT_ENCODING
```
Comment (arthurTemporim) [inline rasa/core/training/interactive.py:822]: Why you are using "r"? Maybe do something like append_write variable in line 819 is better.

**Review by arthurTemporim — CHANGES_REQUESTED** [review]

File: rasa/core/training/interactive.py (hunk 2 above)
Comment (joaoCeilandia) [reply rasa/core/training/interactive.py:822]: In this code we are only reading the file to find the "interactive_story" string. There's no need to use append_write.

File: rasa/core/training/interactive.py (hunk 1 above)
Comment (joaoCeilandia) [reply rasa/core/training/interactive.py:78]: I don't remember doing that. I"ll change it.

**Pull request closed without merging**